### PR TITLE
Automatic forsaken spawns groundside during hijack

### PR DESCRIPTION
**Pull request by morrowwolf**
# About the pull request

This PR automatically spawns forsaken at a 1:1 marine:xeno rate when marines are groundside during hijack.

The xenos are only tier 1/2s. With a 3:1 ratio of tier 1 to tier 2.

# Explain why it's good for the game

The planet is meant to still be an infested hellhole rather than empty once the hive leaves. You *should* fear it.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl: Morrow
add: Added automatic forsaken spawns groundside during hijack
/:cl:

### CODE REVIEW

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/colonialmarines/colonialmarines.dm [hunk 1]
```diff
@@ -248,6 +249,38 @@
...
+			continue
+
+		if(isxeno(current_mob))
+			groundside_xenos++
+			continue
+
+	if(groundside_humans >= (groundside_xenos * GROUNDSIDE_XENO_MULTIPLIER))
+		SSticker.mode.get_specific_call("Xenomorphs Groundside (Forsaken)", FALSE, FALSE)
+
+		for(var/mob/groundside_mob in groundside_humans)
```
Comment (Segrain) [inline code/game/gamemodes/colonialmarines/colonialmarines.dm:277]: That is a number, not a list.

**Comment by MusselMain** [comment]
hopefully these spawns won't extend rounds if marines win the hijack

File: code/datums/emergency_calls/forsaken_xenos.dm [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	name_of_spawn = /obj/effect/landmark/ert_spawns/groundside_xeno
+	objectives = "You have been left behind to safeguard the abandoned colony. Do not allow trespassers."
+
+/datum/emergency_call/forsaken_xenos/spawn_items()
+	var/turf/drop_spawn = get_spawn_point(TRUE)
+	if(istype(drop_spawn))
+		//drop some weeds for xeno plasma regen.
+		new /obj/effect/alien/weeds/node/forsaken(drop_spawn)
+
+/datum/emergency_call/forsaken_xenos/create_member(datum/mind/M, turf/override_spawn_loc)
```
Comment (harryob) [inline code/datums/emergency_calls/forsaken_xenos.dm:15]:
```suggestion
/datum/emergency_call/forsaken_xenos/create_member(datum/mind/new_member, turf/override_spawn_loc)
```

File: code/datums/emergency_calls/forsaken_xenos.dm [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	var/turf/spawn_loc = override_spawn_loc ? override_spawn_loc : get_spawn_point()
+
+	if(!istype(spawn_loc))
+		return //Didn't find a useable spawn point.
+
+	var/mob/current_mob = M.current
```
Comment (harryob) [inline code/datums/emergency_calls/forsaken_xenos.dm:21]:
```suggestion
	var/mob/current_mob = new_member.current
```

File: code/datums/emergency_calls/forsaken_xenos.dm [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	var/mob/living/carbon/xenomorph/new_xeno
+	if(!leader)
+		picked = pick(/mob/living/carbon/xenomorph/warrior, /mob/living/carbon/xenomorph/lurker, /mob/living/carbon/xenomorph/spitter)
+		leader = new_xeno
+	else
+		picked = pick(/mob/living/carbon/xenomorph/drone, /mob/living/carbon/xenomorph/runner)
+
+	new_xeno = new picked(spawn_loc)
+
+	M.transfer_to(new_xeno, TRUE)
```
Comment (harryob) [inline code/datums/emergency_calls/forsaken_xenos.dm:33]:
```suggestion
	new_member.transfer_to(new_xeno, TRUE)
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**